src/transformers/models/intern_s1/tokenization_intern_s1.py

```diff
@@ -0,0 +1,981 @@
+        if split_special_tokens:
+            no_split_token = []
+            tokens = [text]
+        else:
+            no_split_token = self._added_tokens_encoder.keys()  # don't split on any of the added tokens
+            # "This is something<special_token_1>  else"
+            tokens = self.tokens_trie.split(text)
+
+        # ["This is something", "<special_token_1>", "  else"]
+        for i, token in enumerate(tokens):
+            if token in no_split_token:
+                tok_extended = self._added_tokens_decoder.get(self._added_tokens_encoder[token], None)
+                left = tokens[i - 1] if i > 0 else None
+                right = tokens[i + 1] if i < len(tokens) - 1 else None
+                if isinstance(tok_extended, AddedToken):
+                    if tok_extended.rstrip and right:
+                        # A bit counter-intuitive but we strip the left of the string
+                        # since tok_extended.rstrip means the special token is eating all white spaces on its right
+                        tokens[i + 1] = right.lstrip()
+                    # Strip white spaces on the left
+                    if tok_extended.lstrip and left:
+                        tokens[i - 1] = left.rstrip()  # Opposite here
+                    if tok_extended.single_word and left and left[-1] != " ":
+                        tokens[i - 1] += token
+                        tokens[i] = ""
+                    elif tok_extended.single_word and right and right[0] != " ":
+                        tokens[i + 1] = token + tokens[i + 1]
+                        tokens[i] = ""
+                else:
+                    raise ValueError(
+                        f"{tok_extended} cannot be tokenized because it was not properly added"
+                        f" to the tokenizer. This means that it is not an `AddedToken` but a {type(tok_extended)}"
+                    )
+
+        # ["This is something", "<special_token_1>", "else"]
+        tokenized_text = []
+
+        # Code for Auto Detect
+        if self._extra_tokenizer_list is not None:
+            new_tokens = []
+            not_split_flag = 0
+            for token in tokens:
+                if not token:
+                    continue
+                if token in no_split_token or token in self.protect_sp_tokens:
+                    new_tokens.append(token)
+                    if token in self.extra_tokenizer_bos_keys or token in self.protect_begin_sp_tokens:
+                        not_split_flag += 1  # In case nested sp tokens
+                    elif token in self.extra_tokenizer_eos_keys or token in self.protect_end_sp_tokens:
+                        not_split_flag = max(0, not_split_flag - 1)
+                else:
+                    if not_split_flag:
+                        new_tokens.append(token)
+                    else:
+                        for check_module in self.check_module_list:
+                            token = check_module.re_split(token)
```

Comment:
I don't know if performance wise this is the best (but as there are only 2 check module list it should be fine)

Reply:
btw this is one of the most important part right? The module check is what transformers a full string into split tokens. no?

Reply:
I am wondering if it makes sense to for loop, when most probably only 1 of the module will be used, and we hard code to having 2